flake.nix

```diff
@@ -0,0 +1,81 @@
+{
+  inputs = {
+    nixpkgs.url = "github:nixos/nixpkgs/nixos-unstable";
+  };
+
+  outputs = { self, nixpkgs }:
+    let
+      inherit (builtins)
+        substring
+        ;
+      inherit (nixpkgs.lib)
+        genAttrs
+        optionals
+        ;
+
+      eachSystem = f: genAttrs
+        [
+          "aarch64-darwin"
+          "aarch64-linux"
+          "x86_64-darwin"
+          "x86_64-linux"
+        ]
+        (system: f nixpkgs.legacyPackages.${system});
+
+      rev = fallback:
+        if self ? rev then
+          substring 0 8 self.rev
+        else
+          fallback;
+    in
+    {
+      devShells = eachSystem (pkgs: {
+        default = pkgs.mkShell {
+          packages = with pkgs; [
+            cargo
+            clippy
+            rust-analyzer
+            rustc
+            rustfmt
+          ];
+
+          buildInputs = optionals pkgs.stdenv.isDarwin [
+            pkgs.darwin.apple_sdk.frameworks.CoreServices
+          ];
+
+          RUST_SRC_PATH = pkgs.rustPlatform.rustLibSrc;
+        };
+      });
+
+      formatter = eachSystem (pkgs: pkgs.nixpkgs-fmt);
+
+      packages = eachSystem (pkgs: {
+        default = pkgs.rustPlatform.buildRustPackage {
+          pname = "typst";
+          version = rev "00000000";
+
+          src = self;
+
+          cargoLock = {
+            lockFile = ./Cargo.lock;
+            allowBuiltinFetchGit = true;
+          };
+
+          buildInputs = optionals pkgs.stdenv.isDarwin [
+            pkgs.darwin.apple_sdk.frameworks.CoreServices
```

Comment:
```suggestion
            pkgs.darwin.apple_sdk.frameworks.CoreServices
            pkgs.libiconv
```

Reply:
this is not needed since `buildRustPackage` already adds libiconv, both `nix build --system aarch64-darwin` and `nix build --system x86_64-darwin` works for me

Reply:
I'll take your word on it, I only tried the dev shell. thanks!